### ISSUE TITLE
v5 breaks the Git hook

### ISSUE THREAD

**Issue by honzajavorek** [timothycrosley/isort#1294]
I upgraded to v5, my Git hook looks like this (the same as in README, I double checked now):

```python
#!/usr/bin/env python
import sys
from isort.hooks import git_hook

sys.exit(git_hook(strict=True, modify=True))
```

I'm getting the following error:

```
$ git commit
error: unknown change class ' ' in --diff-filter=ACMRTUXB HEAD
Traceback (most recent call last):
  File ".git/hooks/pre-commit", line 5, in <module>
    sys.exit(git_hook(strict=True, modify=True))
  File "/Users/honza/.local/share/virtualenvs/project-kjnSj-Yq/lib/python3.7/site-packages/isort/hooks.py", line 51, in git_hook
    files_modified = get_lines(diff_cmd)
  File "/Users/honza/.local/share/virtualenvs/project-kjnSj-Yq/lib/python3.7/site-packages/isort/hooks.py", line 31, in get_lines
    stdout = get_output(command)
  File "/Users/honza/.local/share/virtualenvs/project-kjnSj-Yq/lib/python3.7/site-packages/isort/hooks.py", line 20, in get_output
    result = subprocess.run(command, stdout=subprocess.PIPE, check=True)  # nosec - trusted input
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'diff-index', '--cached', '--name-only', '--diff-filter=ACMRTUXB HEAD']' returned non-zero exit status 129.
```

**Using:** macOS Mojave (10.14.6), Homebrew, git 2.27.0, python 3.7.7 (I hope the hook won't get executed using the system python 2.7.16)

**Comment by honzajavorek** [comment]
I confirmed `#/usr/bin/env python` leads to the 2.7 python installation, but even when I changed it to `#!/usr/bin/env python3`, which leads to the 3.7, the problem persists.

I'm unsure what the git hook does exactly and what does `--diff-filter=ACMRTUXB HEAD` mean - whether it's a desired value or a bug, I guess I need some guidance here to continue debugging or to provide a fix.

**Comment by honzajavorek** [comment]
I figured out line 50 in `hooks.py` should get changed like this:

```diff
- diff_cmd = ["git", "diff-index", "--cached", "--name-only", "--diff-filter=ACMRTUXB HEAD"]
+ diff_cmd = ["git", "diff-index", "--cached", "--name-only", "--diff-filter=ACMRTUXB", "HEAD"]
```

However, the hook still doesn't work. It correctly passes on commiting `.isort.cfg`, but evaluates `tests/test_models_base.py` as unsorted (although it has been previously sorted by `isort` and that is the very reason why it's modified), sets `errors` to 1, and silently exits the hook with 1 as an exit status code. No print output, no modification made to the file.

**Comment by danaspiegel** [comment]
I'm seeing this too. (OS X 10.15.5, python 3.8.2, git 2.27.0). In addition, we use pyenv to ensure python versions are set properly on a per project basis and I just confirmed that its using the project specific python version.

This completely breaks isort for us. I've held isort back to <5 until fixed.

**Comment by timothycrosley** [comment]
Thanks @honzajavorek and @danaspiegel for raising this!
I've started using the git hook, to hopefully identify any further issues before they become user impacting, and believe I have fixed this in the 5.0.8 release.

Thanks!

~Timothy

**Comment by honzajavorek** [comment]
Awesome, thanks :)
